github/pulls.go

```diff
@@ -197,6 +197,7 @@ type pullRequestUpdate struct {
 }
 
 // Edit a pull request.
+// pull may be nil, in which case the requested PR is returned unmodified.
```

Comment:
I'm trying to think about the best thing to do in a situation where `pull` is `nil`.

Have you also considered returning an error, similar to when `commit` is `nil` in `CreateCommit`?

If so, how did you pick which one to go with?

I can see that for the `CreateCommit` case, there's really no other viable choice but to return error if `commit` is `nil`. You can't create a commit if it weren't supplied, and the only reason `Commit` is being given as a pointer is to copy less memory, not because `nil` value has any meaning.

For `Edit`ing a pull request, it's a little less clear and both options are viable.

My main concern with accepting `pull` value of `nil` as a valid operation is that it would encourage doing that in some situations, but it seems silly to make a network request to GitHub API servers with an noop operation, doesn't it? So maybe we should be more aggressive and return error here too, if `pull` is `nil`?

Or do you think it makes sense to allow the pull request parameter value of `nil`? In that case, should we consider not making the network request, or is that "too smart" for what's meant to be more or less a direct wrapper to the GH API?